### PR TITLE
Docker role refactor

### PR DESCRIPTION
**Pull request by detiber**
- refactors the docker role to push generic config into docker role and wrap
  openshift specific variables into an openshift_docker role and it's
  dependent openshift_docker_facts role
- adds support for setting --confirm-def-push flag (Resolves
  https://github.com/openshift/openshift-ansible/issues/1014)
- moves docker related facts from common/node roles to a new docker role
- renames cli_docker_* role varialbes to openshift_docker-* (maintaining
  backward compatibility)
- update role dependencies to pull in openshift_docker conditionally based on
  is_containerized
- remove playbooks/common/openshift-docker since the docker role is now
  conditionally included

### CODE REVIEW

**Comment by detiber** [comment]
aos-ci-test

**Comment by arilivigni** [comment]
aos-ci-test

**Comment by arilivigni** [comment]
aos-ci-test

**Comment by arilivigni** [comment]
@detiber I reran this with the changes to inventory that you recommended it worked

**Comment by detiber** [comment]
ais-ci-test

**Comment by detiber** [comment]
aos-ci-test

**Comment by detiber** [comment]
@sdodson This PR has the ordering updates to templated systemd unit files.

**Comment by detiber** [comment]
@abutcher mind giving this a review?

**Comment by detiber** [comment]
aos-ci-test

File: roles/openshift_cli/meta/main.yml [hunk 1]
```diff
@@ -12,5 +12,6 @@ galaxy_info:
   categories:
   - cloud
 dependencies:
-- { role: openshift_common }
-- { role: docker }
+- role: openshift_docker
+  when: openshift.common.is_containerized
```
Comment (abutcher) [inline roles/openshift_cli/meta/main.yml]: Needs a bool filter.

File: roles/openshift_etcd/meta/main.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  platforms:
+  - name: EL
+    versions:
+    - 7
+  categories:
+  - cloud
+dependencies:
+- role: openshift_facts
+- role: openshift_docker
+  when: openshift.common.is_containerized
```
Comment (abutcher) [inline roles/openshift_etcd/meta/main.yml]: Bool filter

**Comment by abutcher** [comment]
LGTM aside from those two `is_containerized` needing filters

**Comment by detiber** [comment]
aos-ci-test

**Comment by detiber** [comment]
@brenton this should be good to go now.

**Pull request merged**